controllers/operands/deploymentHandler.go

```diff
@@ -76,5 +71,30 @@ func hasCorrectDeploymentFields(found *appsv1.Deployment, required *appsv1.Deplo
 		reflect.DeepEqual(found.Spec.Template.Spec.Containers, required.Spec.Template.Spec.Containers) &&
 		reflect.DeepEqual(found.Spec.Template.Spec.ServiceAccountName, required.Spec.Template.Spec.ServiceAccountName) &&
 		reflect.DeepEqual(found.Spec.Template.Spec.PriorityClassName, required.Spec.Template.Spec.PriorityClassName)
+}
 
+// updateOrRecreateDeployment either updates or recreates (delete & create) the Deployment based on diff between
+// LabelSelector values of existing vs. required Deployment
+func updateOrRecreateDeployment(Client client.Client, req *common.HcoRequest, found, required *appsv1.Deployment) (bool, bool, error) {
```

Comment:
I was suggesting to have an helper function just to detect if we have to recreate or update

Reply:
Ack. Changed it to reflect what you suggested.